metaci/build/tasks.py

```diff
@@ -134,14 +169,17 @@ def check_queued_build(build_id):
         # For scratch orgs, we don't need concurrency blocking logic,
         # but we need to check capacity
 
+        # TODO: isn't there a very large race condition relating to
+        # checking for scratch org reserve before we even queue the
+        # build on the worker queue?
```

Comment:
When we redo this on top of Pipeline API, I think it would be better to catch the specific failure and re-queue instead of just having a race condition which probably generates unrecoverable failures.